test/src/single_line_comment/parser/highlight_single_line_comment_parser_test.dart

```diff
@@ -0,0 +1,311 @@
+            outerContent: '// comment',
+          ),
+        ],
+      ),
+
+      // ==================================
+      //                Python
+      // ==================================
+
+      _Data(
+        language: python,
+        name: 'Python. Parses unquoted sequences',
+        text: '''
+# hash comment1 
+def fn():
+  text # hash comment2 # still comment2
+''',
+        comments: const [
+          SingleLineComment(
+            lineIndex: 0,
+            innerContent: ' hash comment1 ',
+            outerContent: '# hash comment1 ', // note the trailing space.
+          ),
+          SingleLineComment(
+            lineIndex: 2,
+            innerContent: ' hash comment2 # still comment2',
+            outerContent: '# hash comment2 # still comment2',
+          ),
+        ],
+      ),
+
+      _Data(
+        language: python,
+        name: 'Python. Ignores quoted sequences',
+        text: '''
+def fn(): # comment
+  str1 = '# not a comment';
+  str2 = \'''
+  # not a comment
+  \'''
+  str3 = "# not a comment";
+  str4 = """
+  # not a comment
+  """
+''',
+        comments: const [
+          SingleLineComment(
+            lineIndex: 0,
+            innerContent: ' comment',
+            outerContent: '# comment',
+          ),
+        ],
+      ),
+
+      _Data(
+        language: python,
+        name: 'Python. Breaks on missing colon',
+        text: '''
+def fn() # comment
+''',
+        isLanguageLost: true,
+      ),
+
+      // ==================================
+      //                Scala
+      // ==================================
+
+      _Data(
+        language: scala,
+        name: 'Scala. Parses unquoted sequences',
+        text: '''
+// slashed comment1 
+/* multi
+line */
+public class MyClass {
+  text // slashed comment2 // still comment2
+}
+''',
+        comments: const [
+          SingleLineComment(
+            lineIndex: 0,
+            innerContent: ' slashed comment1 ',
+            outerContent: '// slashed comment1 ', // note the trailing space.
+          ),
+          SingleLineComment(
+            lineIndex: 4,
+            innerContent: ' slashed comment2 // still comment2',
+            outerContent: '// slashed comment2 // still comment2',
+          ),
+        ],
+      ),
+
+      _Data(
+        language: scala,
+        name: 'Scala. Ignores quoted sequences',
+        text: '''
+/*
+ // not a comment
+  */
+public class MyClass { // comment
+  val str3 = "// not a comment";
+}
+''',
+        comments: const [
+          SingleLineComment(
+            lineIndex: 3,
+            innerContent: ' comment',
+            outerContent: '// comment',
+          ),
+        ],
+      ),
+
+      _Data(
+        language: scala,
+        name: 'Scala. Breaks on multiline strings',
+        text: '''
+val str4 = """
+           // not a comment
+           """;
+''',
+        isLanguageLost: true,
+      ),
+    ];
+
+    for (final one in data) {
```

Comment:
is there a better word than "one"? (case, item, etc.)

Reply:
+